data-donation-sdk/src/commonMain/kotlin/care/data4life/datadonation/donation/fhir/validator/ResourceValidator.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2021 D4L data4life gGmbH / All rights reserved.
+ *
+ * D4L owns all legal rights, title and interest in and to the Software Development Kit ("SDK"),
+ * including any intellectual property rights that subsist in the SDK.
+ *
+ * The SDK and its documentation may be accessed and used for viewing/review purposes only.
+ * Any usage of the SDK for other purposes, including usage for the development of
+ * applications/third-party applications shall require the conclusion of a license agreement
+ * between you and D4L.
+ *
+ * If you are interested in licensing the SDK for your own applications/third-party
+ * applications and/or if you’d like to contribute to the development of the SDK, please
+ * contact D4L by email to help@data4life.care.
+ */
+
+package care.data4life.datadonation.donation.fhir.validator
+
+import care.data4life.datadonation.donation.fhir.AllowedReference
+import care.data4life.datadonation.donation.program.model.ProgramFhirResourceBlur
+import care.data4life.hl7.fhir.stu3.model.FhirObservation
+import care.data4life.hl7.fhir.stu3.model.FhirQuestionnaireResponse
+import care.data4life.hl7.fhir.stu3.model.FhirResearchSubject
+import care.data4life.hl7.fhir.stu3.model.FhirResource
+
+internal class ResourceValidator(
```

Comment:
do we actually need a generic resource validator if it's just casting the resource and using all the needed parameters for some of the validation?  feels simpler to either: 
1) export all 3 of them, OR
2) using some sort of enum type, eg.

```
enum ResourceValidatorRule {
   case questionnaireResponse(QuestionnaireResponse, blurMapping)
   case observation(Observation, reference)
   case researchSubject(ResearchSubject, studyId)
}
```

to use in the resource validator like `isAllowed(rule: ResourceValidatorRule)`

Reply:
I'am not entirely getting what you are meaning here, to be honest. We need to pass the resources through in a generic manor until the point we need to know the actual type. I also not getting how enums help here or from where they should come in the end.

Reply:
And it not only casting stuff...it distributes to their specific validator...this a simple [facade](https://en.wikipedia.org/wiki/Facade_pattern).

Reply:
check slack for what i mean in terms of associated Value enum. Facade is not the issue, the issue is its forcing a facade with a non-scalable api